addons/web_editor/static/src/js/wysiwyg/widgets/link.js

```diff
@@ -243,6 +237,22 @@ const Link = Widget.extend({
         }
         return link;
     },
+    /**
+     * Public method to focus the url input.
+     * Called recursively until the input is focused.
+     */
+    async focusUrl() {
+        if (this.$el) {
+            const [$firstInput] = this.$el.find('input:visible:first');
+            if ($firstInput) {
+                $firstInput.focus();
+                $firstInput.select();
+                return;
+            }
+        }
+        await new Promise(resolve => setTimeout(resolve, 50));
+        await this.focusUrl();
```

Comment:
This cannot be the right solution :grimacing: There is code which opens the URL input and put it in the DOM... why can't we focus it if a given option is true at the moment?

Reply:
The link tools are rendered and started in the `Wysiwyg.toggleLinkTools` method, but they are inserted in the DOM and displayed to the user in the `SnippetsMenu._updateRightPanel`. 
Probably that the logic from the current `Wysiwyg.toggleLinkTools` should be split into an `openLinkDialog` that would stay in the Wysiwyg and a new `SnippetsMenu.toggleLinkTools`, as the link tools as such seem very specific the SnippetsMenu. I will push a POC